examples/cluster/plot_inductive_clustering.py

```diff
@@ -0,0 +1,98 @@
+"""
+==============================================
+Inductive Clustering
+==============================================
+
+Clustering is expensive, especially when our dataset contains millions of
+datapoints. Recomputing the clusters everytime we receive some new data
+is thus in many cases, intractable. With more data, there is also the
+possibility of degrading the previous clustering.
+One solution to this problem, is to first infer the target classes using
+some unsupervised learning algorithm and then fit a classifier on the
+inferred targets, treating it as a supervised problem.
+"""
+print(__doc__)
+
+import matplotlib.pyplot as plt
+from sklearn.base import BaseEstimator, clone
+from sklearn.cluster import AgglomerativeClustering
+from sklearn.datasets import make_blobs
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.utils.metaestimators import if_delegate_has_method
+
+
+N_SAMPLES = 5000
+RANDOM_STATE = 42
+
+
+class InductiveClusterer(BaseEstimator):
+    def __init__(self, clusterer, classifier):
+        self.clusterer = clusterer
+        self.classifier = classifier
+
+    def fit(self, X, y=None):
+        self.clusterer_ = clone(self.clusterer)
+        self.classifier_ = clone(self.classifier)
+        y = self.clusterer_.fit_predict(X)
+        self.classifier_.fit(X, y)
+        return self
+
+    @if_delegate_has_method(delegate='classifier')
+    def predict(self, X):
+        return self.classifier_.predict(X)
+
+    @if_delegate_has_method(delegate='classifier')
+    def decision_function(self, X):
+        return self.classifier_.decision_function(X)
+
+
+def plot_scatter(X,  color, alpha=0.5):
+    return plt.scatter(X[:, 0],
+                       X[:, 1],
+                       c=color,
+                       alpha=alpha,
+                       edgecolor='k')
+
+
+# Generate some training data from clustering
+X, y = make_blobs(n_samples=N_SAMPLES,
+                  cluster_std=[1.0, 1.0, 0.5],
+                  centers=[(-5, -5), (0, 0), (5, 5)],
+                  random_state=RANDOM_STATE)
+
+
+# Train a clustering algorithm on the training data and get the cluster labels
+clusterer = AgglomerativeClustering(n_clusters=3)
+cluster_labels = clusterer.fit_predict(X)
+
+plt.subplot(131)
```

Comment:
Please add before this line `plt.figure(figsize=(12, 4))` to specify a figure size.
(and make sure the size is good)

Reply:
It did the trick!